vmbindings/dummyvm/src/tests/fixtures/mod.rs

```diff
@@ -14,15 +17,27 @@ pub trait FixtureContent {
 pub struct Fixture<T: FixtureContent> {
     content: AtomicRefCell<Option<Box<T>>>,
     once: Once,
+    serial_lock: Option<Mutex<()>>,
 }
 
 unsafe impl<T: FixtureContent> Sync for Fixture<T> {}
 
 impl<T: FixtureContent> Fixture<T> {
+    #[allow(dead_code)]
     pub fn new() -> Self {
         Self {
             content: AtomicRefCell::new(None),
             once: Once::new(),
+            serial_lock: None,
+        }
+    }
+
+    #[allow(dead_code)]
+    pub fn new_serial_tests() -> Self {
+        Self {
+            content: AtomicRefCell::new(None),
+            once: Once::new(),
+            serial_lock: Some(Mutex::default()),
```

Comment:
If a fixture is serial, then neither `AtomicRefCell` nor `Once` are necessary. It is easier to create a dedicated `SerialFixture` type, and wrap both `Option<Box<T>>` and an `initialized` flag into a big `Mutex`.